Handle None being returned for ingresses

motjuste opened the pull request:
`svc.status.loadBalancer.ingress` has been observed to return `None` in certain situations and leads to failure of the `ingress-relation-created` hook.  This change handles the `None` and treats the situation as if no ingresses are present.

motjuste commented:
I am happy to write tests for this, but would need guidance to get started.

orfeas-k commented:
Hi @motjuste, does this PR try to resolve https://github.com/canonical/istio-operators/issues/287?

motjuste commented:
> Hi @motjuste, does this PR try to resolve #287?

Not intentionally, but yes, I also faced the same issue.

orfeas-k commented:
That's great. Thank you for taking the time to try and fix this. Could you add a unit test then that replicates the situtation as described in the issue https://github.com/canonical/istio-operators/issues/287#issuecomment-1581196275? This way, we 'll make sure that the fix you 've sent makes what we expect it to.

You could do that by adding another [fixture](https://github.com/canonical/istio-operators/blob/main/charms/istio-pilot/tests/unit/test_charm.py#L1426-L1436) for the specific case and then adding it as a parameter to the tests we already have ([first](https://github.com/canonical/istio-operators/blob/main/charms/istio-pilot/tests/unit/test_charm.py#L535-L546) and [second](https://github.com/canonical/istio-operators/blob/main/charms/istio-pilot/tests/unit/test_charm.py#L506-L517)).

motjuste commented:
Thanks for pointing me to the tests.  I'll update the PR with them.

motjuste commented:
Added a new mock fixture (sorry for the verbose name, suggestions welcome), and then I added it to the two parameterised tests.  

I can't seem to check the CI outputs.  I am new to GitHub Actions ... is there a way for me to get the test running in GitHub without having to set everything up on my laptop?

orfeas-k commented:
Hi @motjuste, we discussed this issue with our team internally and decided that we should start allowing PRs from external forks to run workflows (issue https://github.com/canonical/bundle-kubeflow/issues/764). We will look to implement this in the near future but at the moment, this is not possible (for security reasons as mentioned in the issue). 

For that reason, we will hold this PR until that issue is resolved, so we can properly test and merge it. Once again, thank you for taking the time to contribute to our project.
